### PR TITLE
Add ``log_config`` to config API page

### PR DESCRIPTION
**Pull request by joecummings**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [x] update tests and/or documentation
- [ ] other (please add here)

#### Changelog
* Add ``log_config`` to config API page
* Update docstring for ``log_config``

#### Test plan
EYES

### CODE REVIEW

File: torchtune/config/_utils.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 def log_config(recipe_name: str, cfg: DictConfig) -> None:
     """
-    Logs the parsed config to rank zero.
+    Logs the resolved config to rank zero.
```
Comment (pbontrager) [inline torchtune/config/_utils.py:20]: Can you say what resolved means here? e.g. A resolved config is the provided config updated with any command line overrides.

**Review by pbontrager — COMMENTED** [review]
I looks good. I can't accept this PR though because the docs aren't rendering for me.

File: torchtune/config/__init__.py [hunk 2]
```diff
@@ -10,8 +10,8 @@
 from ._validate import validate
 
 __all__ = [
-    "parse",
     "instantiate",
-    "validate",
+    "parse",
     "log_config",
+    "validate",
```
Comment (RdoubleA) [inline torchtune/config/__init__.py:16]: I wonder if we should just put an assert at the bottom of all our `__init__.py` to ensure this remains alphabeticaly

**Review by RdoubleA — APPROVED** [review]

File: torchtune/config/__init__.py (hunk 2 above)
Comment (joecummings) [reply torchtune/config/__init__.py:16]: I'm sure this is a flake8 command somewhere - I'll investigate.

**Pull request merged**